File: exercises/csharp/TriviaGame/TriviaGame.Core/Board.cs
using System.Collections.Generic;
using System.Linq;

namespace TriviaGame.Core
{
  public class Board
  {
    public static readonly int MAX_QUESTIONS_PER_CATEGORY = 50;
    private readonly Dictionary<string, List<string>> _questions = new Dictionary<string, List<string>>();
    private readonly string[] _categories = { "Pop", "Science", "Sports", "Rock" };
    private string _category(int value) => _categories[value % 4];

    public Board()
    {
      _questions = _categories
        .ToDictionary(
          category => category,
          category => Enumerable
            .Repeat($"{category} Question ", MAX_QUESTIONS_PER_CATEGORY)
            .Select((question, index) => question + index)
            .ToList());
    }

    public string NextQuestion(int boardPosition)
    {
      return _questions[_category(boardPosition)].First();
    }
  }
}

File: exercises/java/src/main/java/com/adaptionsoft/games/Board.java
package com.adaptionsoft.games;

import com.adaptionsoft.games.categories.*;

import java.util.HashMap;
import java.util.Map;

public class Board {

    public Map<Integer, Category> createBoard() {
        Map<Integer, Category> positions = new HashMap<>();
        Pop pop = new Pop();
        Science science = new Science();
        Sports sports = new Sports();
        Rock rock = new Rock();
        positions.put(1, pop);
        positions.put(2, science);
        positions.put(3, sports);
        positions.put(4, rock);
        positions.put(5, pop);
        positions.put(6, science);
        positions.put(7, sports);
        positions.put(8, rock);
        positions.put(9, pop);
        positions.put(10, science);
        positions.put(11, sports);
        positions.put(12, rock);
        return positions;
    }
}
